Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItemDefault.swift

```diff
@@ -0,0 +1,208 @@
+            addSubview(notificationDot)
+        }
+
+        setUp()
+    }
+
+    init(name: String, iconDefinition: IconDefinition) {
+        self.name = name
+        self.icon = iconDefinition
+
+        super.init(frame: .zero)
+
+        setUp()
+    }
+
+    init(name: String, iconDefinition: IconDefinition, showNotificationDot: Bool) {
+        self.name = name
+        self.icon = iconDefinition
+        self.showNotificationDot = showNotificationDot
+
+        super.init(frame: .zero)
+
+        if (showNotificationDot) {
+            addSubview(notificationDot)
+        }
+
+        setUp()
+    }
+
+
+    override var intrinsicContentSize: CGSize {
+        let buttonIntrinsicContentSize = super.intrinsicContentSize
+
+        if let icon = icon {
+            let iconHeight = icon.size.height
+            let width = buttonIntrinsicContentSize.width + 3 * Size.horizontalSpacing
+            let height = iconHeight + Size.verticalSpacing / 2.0
+
+            return CGSize(width: width, height: height)
+        } else {
+            let width = buttonIntrinsicContentSize.width + 2 * Size.horizontalSpacing
+            let height = buttonIntrinsicContentSize.height
+
+            return CGSize(width: width, height: height)
+        }
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        if let frame = titleLabel?.frame, showNotificationDot {
+            let location = notificationLocation(for: frame)
+            self.notificationDot.frame = CGRect(x: location.x, y: location.y, width: BPKSpacingMd, height: BPKSpacingMd)
+        }
+    }
+
+    private func notificationLocation(for labelFrame: CGRect) -> CGPoint {
+        if BPKRTLSupport.viewIsRTL(self) {
+            return CGPoint(x: labelFrame.minX - BPKSpacingMd, y: labelFrame.minY - BPKSpacingSm)
+        } else {
+            return CGPoint(x: labelFrame.maxX, y: labelFrame.minY - BPKSpacingSm)
+        }
+    }
+
+    private func setUp() {
+        self.titleLabel?.lineBreakMode = .byTruncatingTail
+
+        if let icon = icon {
+            setImage(icon.image, for: .normal)
+            adjustsImageWhenHighlighted = false
+            imageView?.tintColor = contentColor
+        }
+
+        updateStyle()
+    }
+
+    private func updateStyle() {
+        let titleString = BPKFont.makeAttributedString(
+            fontStyle: Size.fontStyle, content: name, textColor: contentColor
+        )
+
+        setAttributedTitle(titleString, for: .normal)
+
+        updateInsets()
+        updateIconStyle()
+    }
+
+    private func updateInsets() {
+        if !hasIcon {
+            titleEdgeInsets = UIEdgeInsets(
+                top: Size.verticalSpacing, left: Size.horizontalSpacing,
+                bottom: Size.verticalSpacing, right: Size.horizontalSpacing
+            )
+            return
+        }
+
+        let horizontalInset = Size.horizontalSpacing / 2.0
+
+        if BPKRTLSupport.viewIsRTL(self) {
```

Comment:
as a side comment:
if RTL is used as a configuration point, we can probably think of a configuration protocol that can have 2 or more implementations. This should make the views a bit more agnostic of the user's configuration and have that be configuration of the component. something like this:
```swift
protocol HorizontalNavigationItemDefaultViewConfiguration {
   func imageEdgeInsets(forSize: Size) -> UIEdgeInsets
   func titleEdgeInsets(forSize: Size) -> UIEdgeInsets
   func notificationLocation(fromCoords coords: (x: CGFloat, y: CGFloat)) -> CGPoint
}

class RTL_HNIDefaultViewConfiguration: HorizontalNavigationItemDefaultViewConfiguration {
   func imageEdgeInsets(forSize size: Size) -> UIEdgeInsets {
       return UIEdgeInsets(top: 0, left: size.horizontalSpacing / 2, bottom: 0, right: 0)
   }

   func titleEdgeInsets(forSize size: Size) -> UIEdgeInsets {
      UIEdgeInsets(top: 0, left: 0, bottom: 0, right: size.horizontalSpacing / 2)
   }
   func notificationLocation(fromCoords coords: (x: CGFloat, y: CGFloat)) -> CGPoint {
      return CGPoint(x: coords.x - BPKSpacingMd, y: coords.y - BPKSpacingSm)
   }
}

class NON_RTL_HNIDefaultViewConfiguration: HorizontalNavigationItemDefaultViewConfiguration {
   func imageEdgeInsets(forSize size: Size) -> UIEdgeInsets {
       return UIEdgeInsets(top: 0, left: 0, bottom: 0, right: size.horizontalSpacing / 2)
   }

   func titleEdgeInsets(forSize size: Size) -> UIEdgeInsets {
      UIEdgeInsets(top: 0, left: size.horizontalSpacing / 2, bottom: 0, right: 0)
   }
   func notificationLocation(fromCoords coords: (x: CGFloat, y: CGFloat)) -> CGPoint {
      return CGPoint(x: coords.x, y: coords.y)
   }
}
```

then it would be called like this

```swift
viewConfiguration.notificationlocation(fromCoords: (labelFrame.minx, labelFrame.minY))

imagEdgeinsets = viewConfiguration.imageInsets(forSize: size)
titleEdgeinsets = viewConfiguration.titleInsets(forSize: size)
```
WDYT?

Reply:
oh I see!
Maybe those configuration structs can be internal then, so that all the configuration is centralised in a single point (`private protocol Config` `struct RTLConfig` or `struct LTRConfig`)

Reply:
Yes maybe, but currently it's just a manner of mirroring so it feels like more work than it's worth

Reply:
giving a bit more thought on it (left for debate), I think that this is useful for Skyscanner app, but this being open source, I think this would be forcing users to use RTL components, even if they don't want to.

Reply:
Supporting RTL is part of our offering and open source is only secondary. Consumers can still disable RTL support by forcing LTR if they wish